questionpy/form/_dsl.py

```diff
@@ -147,106 +156,178 @@ def static_text(label: str, text: str, *,
+            right_label=right_label,
+            required=required,
+            help=help,
+            selected=selected,
+            disable_if=_listify(disable_if),
+            hide_if=_listify(hide_if),
+        ),
         bool if not required or disable_if or hide_if else Literal[True],
-        False if not required or disable_if or hide_if else ...
+        False if not required or disable_if or hide_if else ...,
     )
 
 
 @overload
-def radio_group(label: str, enum: Type[_E], *, required: Literal[False] = False,
-                help: Optional[str] = None,
-                disable_if: _ZeroOrMoreConditions = None, hide_if: _ZeroOrMoreConditions = None) -> Optional[_E]:
+def radio_group(
+    label: str,
+    enum: type[_E],
+    *,
+    required: Literal[False] = False,
+    help: str | None = None,
+    disable_if: _ZeroOrMoreConditions = None,
+    hide_if: _ZeroOrMoreConditions = None,
+) -> _E | None:
     pass
 
 
 @overload
-def radio_group(label: str, enum: Type[_E], *, required: Literal[True],
-                help: Optional[str] = None,
-                disable_if: _OneOrMoreConditions, hide_if: _ZeroOrMoreConditions = None) -> Optional[_E]:
+def radio_group(
+    label: str,
+    enum: type[_E],
+    *,
+    required: Literal[True],
+    help: str | None = None,
+    disable_if: _OneOrMoreConditions,
+    hide_if: _ZeroOrMoreConditions = None,
+) -> _E | None:
     pass
 
 
 @overload
-def radio_group(label: str, enum: Type[_E], *, required: Literal[True],
-                help: Optional[str] = None,
-                disable_if: _ZeroOrMoreConditions = None, hide_if: _OneOrMoreConditions) -> Optional[_E]:
+def radio_group(
+    label: str,
+    enum: type[_E],
+    *,
+    required: Literal[True],
+    help: str | None = None,
+    disable_if: _ZeroOrMoreConditions = None,
+    hide_if: _OneOrMoreConditions,
+) -> _E | None:
     pass
 
 
 @overload
-def radio_group(label: str, enum: Type[_E], *, required: Literal[True],
-                help: Optional[str] = None,
-                disable_if: None = None, hide_if: None = None) -> _E:
+def radio_group(
+    label: str,
+    enum: type[_E],
+    *,
+    required: Literal[True],
+    help: str | None = None,
+    disable_if: None = None,
+    hide_if: None = None,
+) -> _E:
     pass
 
 
-def radio_group(label: str, enum: Type[_E], *, required: bool = False,
-                help: Optional[str] = None,
-                disable_if: _ZeroOrMoreConditions = None, hide_if: _ZeroOrMoreConditions = None) -> Any:
+def radio_group(
+    label: str,
+    enum: type[_E],
+    *,
+    required: bool = False,
+    help: str | None = None,
+    disable_if: _ZeroOrMoreConditions = None,
+    hide_if: _ZeroOrMoreConditions = None,
+) -> Any:
     """Adds a group of radio buttons, of which at most one can be selected at a time.
 
     Args:
         label: Text describing the element, shown verbatim.
-        enum: An ``OptionEnum`` subclass containing the available options.
```

Comment:
Stimmt nicht ganz, den Fehler hat ruff unten nochmal gemacht

Reply:
Geht denn etwas kaputt, wenn in den Docstrings gar keine Typen angegeben werden? Das sollte nämlich eigentlich nicht nötig sein, es ist ja alles getypehinted

Reply:
mkdocstrings setzt da halt die type annotations automatisch ein für die generierten Docs. Das funktioniert grundsätzlich gut, ist für die Docs in gewissen Fällen nicht optimal. Daher, denke ich, ist es sinnvolll die types explizit zu setzen für diese Fälle:

z.b. `hidden`

![Screenshot_2024-03-25_12-51-26](https://github.com/questionpy-org/questionpy-sdk/assets/41736868/e6d138f6-0e9a-4d70-987e-f41741ac7eb6)

Wenn man den Code nicht kennt: Was soll hier `_S` sein?

Besser:

![Screenshot_2024-03-25_12-51-58](https://github.com/questionpy-org/questionpy-sdk/assets/41736868/de05e332-12c4-4e7e-ae2d-b94ccb718389)

oder `radio_group`:

![Screenshot_2024-03-25_12-52-27](https://github.com/questionpy-org/questionpy-sdk/assets/41736868/9d41f1c5-e1a2-4404-85df-90f240cace6f)

Was soll `_E` sein?

Besser:

![Screenshot_2024-03-25_12-52-48](https://github.com/questionpy-org/questionpy-sdk/assets/41736868/145bb9fd-d5b3-4346-9e3a-ee5020d87a86)

Das kommuniziert klar, welchen Typ dieser Parameter erwartet. Die privaten Type-Aliase würde in diesem Fall als Implementationsdetails sehen und nicht in die Public API schreiben.

Reply:
(Für `_ZeroOrMoreConditions` hab ich das jetzt noch nicht gemacht, aber auch da sollten wir m.M.n. explizit `Condition | list[Condition] | None` schreiben.)